### PR TITLE
Added a toString, operator == and hashCode to the Quad class.

### PR DESCRIPTION
**Pull request by bramp**
and minor fix to the vector4 toString

### CODE REVIEW

File: lib/src/vector_math/quad.dart [hunk 1]
```diff
@@ -88,4 +88,21 @@ class Quad {
     _point2.add(offset);
     _point3.add(offset);
   }
+
+  /// Returns a printable string
+  @override
+  String toString() => '[0] $_point0\n[1] $_point1\n'
+      '[2] $_point2\n[3] $_point3\n';
+
+  /// Check if two vectors are the same.
```
Comment (spydon) [inline lib/src/vector_math/quad.dart:97]:
```suggestion
  /// Check if two quads are the same.
```

File: lib/src/vector_math_64/quad.dart [hunk 2]
```diff
@@ -88,4 +88,21 @@ class Quad {
     _point2.add(offset);
     _point3.add(offset);
   }
+
+  /// Returns a printable string
+  @override
+  String toString() => '[0] $_point0\n[1] $_point1\n'
+      '[2] $_point2\n[3] $_point3\n';
+
+  /// Check if two vectors are the same.
```
Comment (spydon) [inline lib/src/vector_math_64/quad.dart:97]:
```suggestion
  /// Check if two quads are the same.
```

File: lib/src/vector_math/vector4.dart [hunk 3]
```diff
@@ -119,8 +119,8 @@ class Vector4 implements Vector {
 
   /// Returns a printable string
   @override
-  String toString() => '${_v4storage[0]},${_v4storage[1]},'
-      '${_v4storage[2]},${_v4storage[3]}';
+  String toString() => '[${_v4storage[0]},${_v4storage[1]},'
```
Comment (spydon) [inline lib/src/vector_math/vector4.dart:122]: The one in _64 also needs this fix

**Review by spydon — APPROVED** [review]
Lgtm, except some minor comments

File: lib/src/vector_math/quad.dart (hunk 1 above)
Comment (bramp) [reply lib/src/vector_math/quad.dart:97]: gah sorry, its obvious where I copied and pasted this from :)

File: lib/src/vector_math/vector4.dart (hunk 3 above)
Comment (bramp) [reply lib/src/vector_math/vector4.dart:122]: it was also fixed in my original commit. Please double check it looks good.

**Comment by bramp** [comment]
Thanks, pushed fixes.

File: lib/src/vector_math/vector4.dart (hunk 3 above)
Comment (spydon) [reply lib/src/vector_math/vector4.dart:122]: Ooh, completely missed it when reviewing on the phone. All good then! :)

**Pull request merged**